mongoengine/base/fields.py

```diff
@@ -282,6 +280,19 @@ def _lazy_load_refs(instance, name, ref_values, *, max_depth):
         )
         return documents
 
+    def __set__(self, instance, value):
```

Comment:
FYI Although this could be a healthy thing to do, this is a breaking change and as is, it affects other fields than just ListField(EnumField), for instance ListField(ObjectId) so this is in my opinion something that would require more reflection.

I reworked this to only apply it for EnumField to limit the blast impact and allow this fix to move forward, I opened another PR here https://github.com/MongoEngine/mongoengine/pull/2722